WordPress/openverse
# Restructure API to remove confusion about 'catalog'

## Fixes



Fixes #1799 by @zackkrida 

## Description



This PR restructure the API to remove the `catalog/` name from the codebase and reduce confusion with the top-level `catalog/` directory.

As per this change, the `api/catalog/` directory has been renamed to `api/conf/` (since it's mostly concerned configuration for the Django project) and the `api/catalog/api/` directory has been lifted up as `api/api/` (the name 'api' was retained so that all existing migrations, table_names etc. continue to remain valid).

## Testing Instructions


CI passes so that means no functionality has been impacted. If you know of any functionality in the API that is not covered by tests, please test that out, and file an issue so that we can add test coverage for it (see  #1657).

Please update the settings module in the `.env` file to `conf.settings` instead of `catalog.settings`.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
